### PR TITLE
The Emag of the People

### PR DESCRIPTION
**Pull request by GreatBigFailure**
[dnm]

tl;dr: one use emag costs 2TC, current emag costs 16. Don't merge it'll explode.

It's time again to talk about the emag. 
This PR is just to have a vote and discussion. Do not merge this unless the vote has decided it. In order to prevent merging, and because it made me smile, I have added these lines:

	to_chat(world, "I HAVE BEEN MERGED")
	sleep(3 SECONDS)
	explosion(get_turf(user),500,500,500,500,1,0)

Which destroys the universe, ignoring bomb cap, upon the new emag being used. 
I have also intentionally made this in the sloppiest, laziest way possible. I tested it a single time and it didn't work quite right then I made this anyway. 

### The numag

Emag detractors complain that the item is too powerful and flexible for its 6TC cost. Even if you don't think the item should be changed you have to admit, it's very cheap. It is however also very fun. Fun is priority. Balance should serve fun.
Way I see it people who buy the emag have one of two reasons: 

1. They have a gimmick that requires an emagged cloner, cargo computer, etc. This requires a single swipe of the emag then it's a liability. It's just evidence.

2. They want to emag 4 borgs, every door on the station, and whatever else they can find. For this 6TC is nothing and leaves them with 14 useless TC.

Why have one item trying to compromise between both? 
What I've done is very lazily made a new emag that turns into a normal blank ID card upon being used. It costs 2TC. I've bumped the current emag, identical to how it was before, to 16TC. I am not married to these numbers. I picked 16 specifically because it primarily allows for stealthy, gimmicky items and also clowns to glue it to their hands. 

**Why would you do this awful thing**
If you want to make a MANCROWAVE or order naughty cargo stuff this is a buff. It costs you 4 less TC and mostly destroys the evidence. 
If you want to open 30 lockboxes and create an army of borgs this is a nerf but only to what you can get in addition to the emag, the emag itself still works on both. 

Again, this is an emag buff until you want to use it 4 times, even at 3 uses it's still a buff. 

### Screech your vote
Originally I took a vote in the thread. These are the results:
https://strawpoll.com/r9wdsdeh/r

Most people didn't want it. We all know that isn't the best place for votes or civil discourse though. So let's talk about it here and vote here but also keep that strawpoll in mind, not everyone wants to bother making a git account.

I am sorry Edward Emaghands.

:cl:
 * rscadd: Welcome the days of one use emag for 2TC, infinite for 16TC 
 * rscadd: Universe destroying explosion
 * rscdel: Gone is the infinite emag for 6TC

### CODE REVIEW

File: code/game/objects/items/weapons/cards_ids.dm [hunk 1]
```diff
@@ -180,6 +180,24 @@
...
+
+/obj/item/weapon/card/emag/afterattack(var/atom/target, mob/user, proximity)
+	..()
+	to_chat(world, "I HAVE BEEN MERGED")
+	sleep(3 SECONDS)
+	explosion(get_turf(user),500,500,500,500,1,0)
+	to_chat(user, "The magnetic strip tears away")
+	user.drop_item(src, force_drop=1)
+	var/obj/item/weapon/card/id/I = new (get_turf(user))
+	qdel(src)
```
Comment (jknpj) [inline code/game/objects/items/weapons/cards_ids.dm:198]:
```suggestion
```
:smile:

**Comment by Irradiation** [comment]
>Strawpoll

That's what server polls are for.

**Comment by Accipitermaris** [comment]
unfortunately i gotta oppose because if i (for example) emag the cloner with my 1 use emag, it can be trivially fixed by just screwdrivering the console. emagging computers to upgrade/sabotage is very weird and can be instantly undone

**Comment by Buchse** [comment]
it makes sense, fluff-wise emags (current) have a generator attached to keep them working. This one's just an emag sans generator.

**Comment by GreatBigFailure** [comment]
> unfortunately i gotta oppose because if i (for example) emag the cloner with my 1 use emag, it can be trivially fixed by just screwdrivering the console. emagging computers to upgrade/sabotage is very weird and can be instantly undone

That's fair. Way I figure it for 2TC you can afford to just re-mag it though. Besides it also gives you away and makes it harder to emag it again same as the current emag. If your cloner has been repaired 3 times and you're on your 4th, still not having been caught, then you probably don't need to worry about TC.

**Comment by BomberBro** [comment]
I support a change to the emag since it currently carries with it a decade of bloat for random machines that has steadily powercreeped it, but I'd like a simpler solution as this is pretty drastic and guts gimmicks using the emag. Originally emags had around 5 uses and recharged a use every minute or so iirc, but this was broken for like five years before anyone noticed it and the PR was made to fix them last year, but it specifically kept traitor emags infinite. I think re-adding the recharge would help to cut down on it being a tool for instant gunk without being too severe, but I don't know how much it'd truly change.

**Comment by Irradiation** [comment]
(Bomber's suggestion is a better alternative to the changes you're proposing)

I frankly don't believe that the average Emag purchaser only emags things 1-2 times and calls it a day. On top of that, the Emag is generally a companion/utility tool that goes with other loadouts. With a doubled (16 TC) cost, you can't buy, say, a Revolver/Energy Sword/Chef Butcher Cleaver/Whatever high-TC-cost-job item you'd want.

tl;dr Nerfing the cost heavily limits your gimmick budget to Emag-only and that is the big gay.

**Comment by GreatBigFailure** [comment]
>I think re-adding the recharge would help to cut down on it being a tool for instant gunk without being too severe, but I don't know how much it'd truly change.

I had considered that for this. There's two problems though:
1. It wouldn't change much. If you buy an emag how many things do you actually emag? Personally it's 1-5, roughly estimating. There was one time I hacked a bunch of door's bolts down then emagged them but that was the only time it was that many. A minute recharge, even 10 minutes on something with 5, or even 3ish, uses wouldn't do anything
2. Default machine variable seems to be emag_cost = 0. Originally I wrote "two problems" intending to say you'd need to update all the machines but now I realized you could just change a single variable to 1. I'm keeping this in though.

File: code/game/objects/items/weapons/cards_ids.dm (hunk 1 above)
Comment (GreatBigFailure) [reply code/game/objects/items/weapons/cards_ids.dm:198]: I put very, very little effort into this

**Comment by GreatBigFailure** [comment]
>Emag is generally a companion/utility tool that goes with other loadouts. With a doubled (16 TC) cost, you can't buy..

These are just spitball numbers. I gave the list a quick ctrl+f for "= 4" and said "Ya 16 whatever". I mostly made this just for discussion rather than to actually merge in its current state. 
I'm also pretty sure I accidentally rigged the normal emag to destroy the universe, not just the new one. 
If it was to be raised, what do you think would be a good number?

**Comment by Irradiation** [comment]
I don't think it should go over 8 TC if we're spitballing actual numbers, lest you can't buy your meme items.

**Comment by GreatBigFailure** [comment]
What items specifically make you say 8? 
TC balance should be based around what is affordable in combination rather than actual price. 8TC and 7TC are identical when thinking about something combining with 12, for example. Not considering random item, I suppose.

**Comment by Kharkov72** [comment]
@DamianX merging this when

**Comment by kilozombie** [comment]
Legacy: Back in the past, emags used to have a random number of uses. They'd basically shut off after a few emags and then be useless.
They have evolved a very long way since then, but they were still used when they were finicky. Introducing limited-use emags isn't going to destroy the balance, but /vg/ is also different now than it was before. People are, on average, stupendously more robust and capable of fixing damage. If you want limited-use emags for specific gimmicks, you should make it- for instance- far harder to fix computers being emagged.
It has always irked me how giga-gunked you can make a round just by emagging every door everywhere. As annoying as it was, finicky limited-use emags made that totally unviable.

**Comment by ghost** [comment]
I've talked with Barry about emag balance in the past, and I think this is absolutely retarded. What would make way more sense and be a good starting point for balancing the emag without pissing people off would be to bring back its recharging behavior but have a fair ratio of starting charges to recharge time. Give it 5-10 uses max and have it recharge one every 5-10 minutes. That way people can use it for fuckery without breaking every door on the station repeatedly.

**Comment by DamianX** [comment]
>People are, on average, stupendously more robust and capable of fixing damage.

I have absolutely no horse in this race, I just found this outstandingly funny.

**Comment by kilozombie** [comment]
> > People are, on average, stupendously more robust and capable of fixing damage.
> 
> I have absolutely no horse in this race, I just found this outstandingly funny.

This is in comparison to how it used to be, and how it is in other communities. /vg/ is extraordinary in terms of people actually knowing what they're doing.
Other servers, people never even lift a finger to fix damage or coordinate departments. It is EXTREMELY unlikely you will be cloned or even healed by Medbay. The fact that people actually know to unscrew computers to fix them is wack in comparison to the average, and if it's true that people commonly do this, then that's amazing.

**Comment by tuypo1** [comment]
Emags don't need to be worse or more expensive, everything else in the uplink needs to be cheaper or better

**Comment by ghost** [comment]
Just give us a one-use emag for 2 TC, keep the normal emag function and pricing the same.

**Comment by GreatBigFailure** [comment]
I realize I presented this more as a discussion of my idea on how to change it rather than in general. That wasn't the intention, just figured I might as well kill two birds with one stone. 
I think it's safe to say the vast majority are against the specific "fix" I've proposed, I'll leave this up for a bit longer just for discussion's sake but close the pull sooner rather than later.

File: code/game/objects/items/weapons/cards_ids.dm (hunk 1 above)
Comment (SonixApache) [reply code/game/objects/items/weapons/cards_ids.dm:198]: next time just make it qdel(world) on init
